staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1035,24 +1035,37 @@ func (e *Store) DeleteCollection(ctx context.Context, deleteValidation rest.Vali
 			for index := range toProcess {
 				accessor, err := meta.Accessor(items[index])
 				if err != nil {
-					errs <- err
-					return
+					select {
+					case errs <- err:
+					default:
+					}
+					continue
 				}
 				if _, _, err := e.Delete(ctx, accessor.GetName(), deleteValidation, options); err != nil && !apierrors.IsNotFound(err) {
 					klog.V(4).Infof("Delete %s in DeleteCollection failed: %v", accessor.GetName(), err)
 					errs <- err
-					return
 				}
 			}
 		}()
 	}
 	wg.Wait()
-	select {
-	case err := <-errs:
-		return nil, err
-	default:
-		return listObj, nil
+	errorsSeen := 0
```

Comment:
collecting the errors after all goroutines have finished requires sizing the errors channel assuming everything fails. I don't think that's a good idea.